packages/flutter/test/material/dropdown_test.dart

```diff
@@ -777,4 +777,48 @@ void main() {
     ), ignoreId: true, ignoreRect: true, ignoreTransform: true));
     semantics.dispose();
   });
+
+  testWidgets('Dropdown listview scroll position when placed not at the top', (WidgetTester tester) async {
+    int value = 0;
+    final List<DropdownMenuItem<int>> items = <DropdownMenuItem<int>>[];
+    for (int i = 0; i < 222; ++i)
+      items.add(DropdownMenuItem<int>(value: i, child: Text('$i')));
+
+    void handleChanged(int newValue) {
+      value = newValue;
+    }
+
+    final DropdownButton<int> button = DropdownButton<int>(
+      value: value,
+      onChanged: handleChanged,
+      items: items,
+    );
+
+    double getMenuScroll() {
+      double scrollPosition;
+      final ListView listView = tester.element(find.byType(ListView)).widget;
+      final ScrollController scrollController = listView.controller;
+      assert(scrollController != null);
+      scrollPosition = scrollController.position.pixels;
+      assert(scrollPosition != null);
+      return scrollPosition;
+    }
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: Align(
+            alignment: Alignment.center,
+            child: button,
+          ),
+        ),
+      ),
+    );
+
+    await tester.tap(find.text('0'));
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1));
+
+    expect(getMenuScroll(), 0.0);
+  });
```

Comment:
let's have some tests for when the button is at the top of the screen but showing the bottom item, and at the bottom of the screen but showing the top item, and in the middle but showing the middle item, and in the middle but showing the bottom item, as well, while we're at it. Cover our bases and make sure everything works so we don't have to deal with it again...

Reply:
Added them!